dev-client/src/hooks/useMapSuggestions.ts

```diff
@@ -0,0 +1,95 @@
+import {useCallback, useState} from 'react';
+
+import {useDebouncedCallback} from 'use-debounce';
+
+import {Coords} from 'terraso-client-shared/types';
+
+import {MAP_QUERY_MIN_LENGTH} from 'terraso-mobile-client/constants';
+import {
+  initMapSearch,
+  Suggestion,
+} from 'terraso-mobile-client/services/mapSearchService';
+import {isValidCoordinates} from 'terraso-mobile-client/util';
+
+/*
+ * Logic extracted from screens/HomeScreen/components/MapSearch.tsx; see its revisions
+ * for history of original implementation / TODOs etc
+ */
+
+const {getSuggestions, retrieveFeature} = initMapSearch();
+
+export const useMapSuggestions = () => {
+  const [coords, setCoords] = useState<Coords | undefined>(undefined);
+  const [suggestions, setSuggestions] = useState<Suggestion[]>([]);
+
+  /*
+   * We use the AbortController API to cancel pending calls when new ones go out;
+   * otherwise a stale async call could overwrite the results of a newer one if it
+   * completes first.
+   */
+  const [, setAbortController] = useState<AbortController | null>(null);
+  const makeSuggestionsApiCall = useCallback(async (queryText: string) => {
+    const newAbortController = new AbortController();
+    setAbortController(current => {
+      if (current !== null) {
+        current.abort();
+      }
+      return newAbortController;
+    });
+    const newSuggestions = await getSuggestions(
+      queryText,
+      newAbortController.signal,
+    );
+    setAbortController(null);
+    return newSuggestions;
+  }, []);
+
+  const querySuggestions = useCallback(
+    async (queryText: string) => {
+      if (isValidCoordinates(queryText)) {
+        const [latitude, longitude] = queryText.split(',').map(Number);
```

Comment:
suggestion: validate the coordinates before submitting them. (we have `isValidLatitude`/ `isValidLongitude` helpers already).

but: not sure what to do if they're invalid. Ask @CourtneyLee333?

This also is somewhat out of scope, so we could do that work in another PR.

Reply:
I don't understand your suggestion - we're invoking `isValidCoordinates` right here which in turn invokes `isValidLatitude(latitude) && isValidLongitude(longitude)`.

Reply:
Oops.